### PR TITLE
Add a custom global whitelist

### LINKED ISSUE #3782: Use a custom whitelist of globals allowed in the source
https://github.com/desktop/desktop/pull/3780 happened because `window.name` is a string property of `window`, which is used for <sub>*mumble mumble*</sub>. However, we probably don’t need to use any of the properties of `window` (or even `window` — `global` should be enough).

We could check for this error with ESLint’s `no-undef` rule, and whitelisting globals as needed (or just accessing them under `global`).

### PR DESCRIPTION
**Pull request by j-f1**
Feedback on the list in `.eslintrc.js` is appreciated.

Fixes #3782.

### CODE REVIEW

**Comment by j-f1** [comment]
💻

**Comment by nerdneha** [comment]
@j-f1 looks like we still have some failing tests in circleci; are you able to access?

File: app/src/lib/source-map-support.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ function retrieveSourceMap(source: string) {
 
   if (__DEV__ && path.startsWith('http://')) {
     try {
-      const xhr = new XMLHttpRequest()
+      const xhr = new window.XMLHttpRequest()
```
Comment (shiftkey) [inline app/src/lib/source-map-support.ts]: I'm not quite sure this is the best approach, as these changes now cause the TS compiler to consider these errors. I'm also not across how best to mash tsc and eslint together at this point, so I'm not sure how to proceed...

**Comment by shiftkey** [comment]
Dropping this from the 1.2 milestone as it's not related to feature work.

**Comment by shiftkey** [comment]
I had another shot at trying to get TS to play nicer with the changes here, and couldn't really figure out a way to mesh this together with the ESLint work here.

@j-f1 do you mind if we close this PR and keep #3782 open to track the desired goal?

**Comment by j-f1** [comment]
Sounds fine

**Pull request closed without merging**